### PR TITLE
Fix soft spi bugs

### PR DESCRIPTION
**Pull request by Enoch247**
### Contribution description

This PR contains some small bug and spelling fixes I noticed while reading the soft_spi code while using it as an example for writing my own driver. See the commit comments for details of each change.

### Testing procedure

I compiled, but did not run, tests/driver_soft_spi. The change is minor and verifiable by simply looking at the code.

### Issues/PRs references

- None

### CODE REVIEW

**Review by maribu — APPROVED** [review]

**Comment by maribu** [comment]
bors merge

Thx for the fix :)

File: drivers/soft_spi/include/soft_spi_params.h [hunk 1]
```diff
@@ -42,12 +42,14 @@ extern "C" {
...
 
 /**
- * @brief   Sotware SPI port descriptor array
+ * @brief   Software SPI port descriptor array
  */
 static soft_spi_conf_t soft_spi_config[] = {
     SOFT_SPI_PARAMS,
 };
 
+#define SOFT_SPI_NUMOF             ARRAY_SIZE(soft_spi_config)
```
Comment (maribu) [inline drivers/soft_spi/include/soft_spi_params.h:51]:
```suggestion
/**
 * @brief   Number of software SPI buses
 */
#define SOFT_SPI_NUMOF             ARRAY_SIZE(soft_spi_config)
```

This should get the doc check happy

Comment (Enoch247) [reply drivers/soft_spi/include/soft_spi_params.h:54]: done

**Pull request merged**